### ISSUE TITLE
.Net 5 Backward Compatibility with earlier versions of .Net Core

### ISSUE THREAD

**Issue by pfdsilva** [dotnet/runtime#35556]
This is not really an issue but a request for information.

Our .Net Core product (ADO.NET driver) is compiled using <TargetFramework>netstandard2.0</TargetFramework>

The global.json file has the following setting.
{
  "sdk": {
    "version": "3.1.103"
  }
}

Since our product is compiled using .Net Core SDK 3.1.103, I am wondering whether customer applications that use our product and which target .Net 5 will run into any issues. 

What if the customer environment has only .Net 5 runtime installed. Will they encounter any issues when they use our product which has been compiled using .Net Core SDK 3.1.103?

To summarize, the question is whether .Net 5 is backward compatible with earlier versions of .Net Core.

**Comment by Gnbrkm41** [comment]
New major versions of .NET Core can and usually do have some breaking changes; for example, [there was a few changes related with parsing/formatting of floating point numbers, to make it IEEE-754 compatible](https://devblogs.microsoft.com/dotnet/floating-point-parsing-and-formatting-improvements-in-net-core-3-0/); while it made sense to make those changes (because "it's the right thing") it does mean different behaviour when compared to Core 2.2 or lower versions.

So it is possible that you may run into problems when running programs compiled for lower versions of runtime on new major versions of runtime. I don't even think .NET Core supports running older programs on runtimes of newer major versions....

**Comment by danmosemsft** [comment]
We try hard to be backward compatible, and when there are known breaks we try to make them during major version increments. In the vast majority of cases we expect code to run fine on the newer version (although, my knowledge is mostly of this level of the stack and less of ASP.NET). We generally assume that apps (rather than libraries) will be recompiled, although that isn't necessarily required.

@ericstj anything to add?

**Comment by ericstj** [comment]
> @ericstj anything to add?

In particular this is .NETStandard2.0 library and the question was around using a 3.1 SDK to compile this library.  You won't see any differences when compiling for netstandard2.0 with the 3.1 SDK vs the 5.0 SDK.  .Net 5 remains .NETStandard2.0 compatible so your library will continue to work in .Net 5.

**Comment by danmosemsft** [comment]
I think this is answered. If you do hit a problem, please open a new issue here.
